src/provisioner/validators.py

```diff
@@ -150,23 +138,24 @@ def _check_external_services(self):
 
         sol_conf_ext_services = self._get_value_from_solution_config(
             ConfigValidator._key_ext_service)
-        conf_stor_ext_services = self._get_value_from_conf_store(
-            ConfigValidator._key_ext_service)
-
-        if len(conf_stor_ext_services) != len(sol_conf_ext_services):
-            raise CortxProvisionerError(errno.EINVAL,
-                'External services define in conf_store is not same as '
-                f'solution_config. External services in {self.solution_conf_url}'
-                f' is {len(sol_conf_ext_services)} and in {self.cortx_conf_url}'
-                f' is {len(conf_stor_ext_services)}')
-
-        # Check endpoints
-        for service in conf_stor_ext_services.keys():
-            if (conf_stor_ext_services[service]['endpoints'] !=
-                    sol_conf_ext_services[service]['endpoints']):
+        for service in sol_conf_ext_services:
+            if self._get_value_from_conf_store(
+                f'{ConfigValidator._key_ext_service}>{service}>admin') is None:
                 raise CortxProvisionerError(errno.EINVAL,
-                    f'{service} endpoints define in conf_store is not same as '
-                    'solution_config.')
+                    'External service {service} is not present in conf_store.')
+            counter = 0
+            endpoints_in_confstore = []
+            while self.cortx_config_store.get(
+                f'{ConfigValidator._key_ext_service}>{service}>endpoints[{counter}]') is not None:
+                endpoints_in_confstore.append(self._get_value_from_conf_store(
+                    f'{ConfigValidator._key_ext_service}>{service}>endpoints[{counter}]'))
+                counter = counter + 1
+        # Check endpoints
+        if (endpoints_in_confstore != sol_conf_ext_services[
```

Comment:
sol_conf_ext_services is misleading. There is no solution conf that is passed to provisioner. config passed is cortx conf and cluster conf. Can we reflect the correct name here